src/scripts/preparation/Merge_data.R

```diff
@@ -0,0 +1,278 @@
+  
+  # Check if there are unmatched rows and increment the counter
+  # if (nrow(unmatched_rows) > 0) {
+  #   files_with_unmatched_rows <- files_with_unmatched_rows + 1
+  #   print(paste("Unmatched rows in file:", basename(csv_file_path)))
+  #   print(unmatched_rows)
+  # }
+  
+  save_path <- gsub(".csv", "_merged.csv", csv_file_path)
+  write.csv(merged_data, save_path, row.names = FALSE, quote = FALSE)
+}
+
+
+
+### =========================================================================
+### E - Create a Comprehensive Name-Matching Table
+### =========================================================================
+
+## Load the data and normalize names for matching
+statistic_data <- read.csv(csv_file_path) %>%
+  mutate(
+    DEPARTMENT_NORM = iconv(DEPARTMENT, "UTF-8", "ASCII//TRANSLIT"),
+    PROVINCE_NORM = iconv(PROVINCE, "UTF-8", "ASCII//TRANSLIT"),
+    DISTRICT_NORM = iconv(DISTRICT, "UTF-8", "ASCII//TRANSLIT")
+  )
+
+district_shapefile <- st_read(shapefile_path) %>%
+  st_drop_geometry() %>% # Drop the geometry column
+  mutate(
+    DEPARTAMEN_NORM = iconv(DEPARTAMEN, "UTF-8", "ASCII//TRANSLIT"),
+    PROVINCIA_NORM = iconv(PROVINCIA, "UTF-8", "ASCII//TRANSLIT"),
+    DISTRITO_NORM = iconv(DISTRITO, "UTF-8", "ASCII//TRANSLIT")
+  )
+
+
+# Perform matching on normalized names
+matched_data <- merge(
+  x = district_shapefile, y = statistic_data,
+  by.x = c("DEPARTAMEN_NORM", "PROVINCIA_NORM", "DISTRITO_NORM"),
+  by.y = c("DEPARTMENT_NORM", "PROVINCE_NORM", "DISTRICT_NORM")
+)
+
+
+# Create the name-matching table using original names
+name_matching_table_full <- matched_data %>%
+  select(
+    UBIGEO_DEPT = IDDPTO,
+    SHP_DEPT = DEPARTAMEN,
+    UBIGEO_PROV = IDPROV,
+    SHP_PROV = PROVINCIA,
+    UBIGEO_DIST = IDDIST,
+    SHP_DIST = DISTRITO,
+    EXL_DEPT = DEPARTMENT,
+    EXL_PROV = PROVINCE,
+    EXL_DIST = DISTRICT,
+    DEPT_ASCII = DEPARTAMEN_NORM,
+    PROV_ASCII = PROVINCIA_NORM,
+    DIST_ASCII = DISTRITO_NORM
+    
+  )
+
+
+# Identify unmatched rows from shapefile data
+unmatched_shapefile_rows <- anti_join(district_shapefile, matched_data, by = c("DEPARTAMEN", "PROVINCIA", "DISTRITO"))
+
+# Create a dataframe for unmatched rows with the required structure
+unmatched_rows_df <- unmatched_shapefile_rows %>%
+  select(
+    UBIGEO_DEPT = IDDPTO,
+    SHP_DEPT = DEPARTAMEN,
+    UBIGEO_PROV = IDPROV,
+    SHP_PROV = PROVINCIA,
+    UBIGEO_DIST = IDDIST,
+    SHP_DIST = DISTRITO,
+    DEPT_ASCII = DEPARTAMEN_NORM,
+    PROV_ASCII = PROVINCIA_NORM,
+    DIST_ASCII = DISTRITO_NORM
+  ) %>%
+  mutate(
+    EXL_DEPT = NA_character_,
+    EXL_PROV = NA_character_,
+    EXL_DIST = NA_character_
+  )
+
+# Append the unmatched rows dataframe to the existing name_matching_table
+name_matching_table_full <- rbind(name_matching_table_full, unmatched_rows_df)
+
+print(name_matching_table_full)
+
+# save_path <- paste0(Data_dir, "/Preds/Raw/INEI/name_matching_table_full.csv")
```

Comment:
Let's save the table at a higher level in the git project `src/utls/Admin_unit_matching_table.csv`